src/libsyntax/parse/token.rs

```diff
@@ -590,9 +584,7 @@ impl Token {
                                                        a.name == keywords::DollarCrate.name() ||
                                                        c.name == keywords::DollarCrate.name()),
 
-            (&Literal(ref a, b), &Literal(ref c, d)) => {
-                b == d && a.probably_equal_for_proc_macro(c)
-            }
+            (&Literal(a, b), &Literal(c, d)) => b == d && a == c,
```

Comment:
Let's flip `&&` if there's no particular reason to compare second components first.  If there is a reason, let's document it?

Reply:
This is also fixed in a next commit (b and d are removed).